Hw15/ObjectCounter.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+template<class T>
```

Comment:
don't really see the need of template here...
Could you explain plz?)

Reply:
I did not want to mix Rectangle's class implementation with counting objects.
I do not remember where I have seen an approach like this, I assume it was S. Meyers

Reply:
It should be made via templates if we are going to use the same ObjectCounter for other objects because if it is implemented like a simple class, it will count how many times ObjectCounter was created, instead of Derived classes(Rectangle, some other shapes).